help-docs: Add a note about read receipts to 'Mute a user' article.

laurynmm opened the pull request:
Adds a bullet point about muted users being excluded from read receipts for all messages.

Follow-up to #22941.

---

**Screenshots and screen captures:**

![Screenshot from 2022-09-28 13-10-23](https://user-images.githubusercontent.com/63245456/192804126-967a10c1-5fbf-42eb-8da5-951bcc006c88.png)

---

**Self-review checklist**

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/version-control.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.

timabbott commented:
Do we need a sentence about whether muted users can see whether you've read messages? We do hide your read receipts information from them, and a good way to frame it might be something like "Muted users cannot tell whether you've read messages; it will be indistinguishable for them from your having set the privacy setting to not share read receipts with other users" with a link.

(Not edited at all, but I think the reference to the privacy setting is helpful).

alya commented:
I dunno, it felt like such an edge case to me, that it seemed more confusing to mention it than not.

alya commented:
Anyway, based on the screenshot above, I'm happy with the proposed changes.

laurynmm commented:
I agree that it feels a bit confusing to mention. But I also could see the first bullet point about marking messages as read could generate concern about the read receipts reflecting that?

* For you, muted users are excluded from [read receipts](https://zulip.com/help/read-receipts) for all messages. For them, read receipts for all messages will be as if you have enabled [the privacy setting](https://zulip.com/help/read-receipts#configure-whether-zulip-lets-others-see-when-youve-read-messages) to not share read receipts with other users.

*Side note*: We should probably link to read receipts as a related article on [this article](https://zulip.com/help/marking-messages-as-read)?

timabbott commented:
Yeah that proposal lgtm

alya commented:
How about this?

> Muted users are excluded from [read receipts](https://zulip.com/help/read-receipts) for all messages. Zulip never shares whether or not you have read a message with a user you've muted.

laurynmm commented:
**Updates**:
- Updated text in first commit to Alya's version of the text above, see screenshot below.
- Added 2nd commit with the addition to the **Related articles** section for the "Marking messages as read" article.

**Screenshots**:
<details>
<summary>Mute a user - updated</summary>

[Current documentation](https://zulip.com/help/marking-messages-as-read)
![Screenshot from 2022-09-30 12-34-46](https://user-images.githubusercontent.com/63245456/193254801-f2c1aa5e-1dd3-4e1a-b071-bdc0bf49355e.png)

</details>

alya commented:
Works for me! @timabbott over to you.
